### ISSUE TITLE
Sending mail via mailbox.org on mobile connection fails with 501 SMTP error

### ISSUE THREAD

**Issue by titusstahl** [grote/BlitzMail#24]
I have a mail account at mailbox.org. When I am using BlitzMail on my mobile connection (as opposed to my home wlan), I always get the following error message:

`Der Server antwortete mit folgendem Fehlder: Bitte überprüfen Sie die SMTP-Einstellungen. 501 Syntax: HELO hostname`

Sending mail via k-9 works with the same account on any connection.

This is on version 0.6

**Comment by grote** [comment]
Is it possible that this is a duplicate of issue #21?

**Comment by titusstahl** [comment]
I don't believe so. In my case, the error message is really consistent, retrying does not make any difference. It's also a different message.

I've written now to the mailbox.org support, asking them to contribute any information they could have about the issue.

**Comment by grote** [comment]
That would be nice. When they have the exact time of your connection attempt, they should be able to get more information about why the message was rejected from their web server logs.

**Comment by deqoksijnc** [comment]
I did recently have problems with another provider and HELO, seems more servers are now checking that HELO is something what they consider "meaningful".  Since mobile networks mostly have transparent proxies and NAT this is quite easy to get wrong:(

**Comment by grote** [comment]
This is also related to #25. Maybe somebody wants to figure out a way for how to get the proper hostname to use in the HELO message?

**Comment by duckdalbe** [comment]
I have the same problem with my own mail-server. Debugging via the MTA showed me that BlitzMail apparently sends only `EHLO` without any argument. That finding contradicts #25.

Does BlitzMail actually send `EHLO` without argument? Or might this be due to network operators filtering headers? It happends also when I tell BlitzMail to use TLS and authentication credentials, though.

**Comment by grote** [comment]
> Does BlitzMail actually send EHLO without argument?

BlitzMail uses libraries to do the sending. I am not sure what it actually sends. I'd be happy for anyone to debug this further.

**Comment by rgh36167** [comment]
The library should do the right thing by default and send the localhostname. It does so only erratically (at least in my case). Since DNS is asyncronous this looks like a timing issue.

It works almost always when I invoke apps/BlitzMail send queued. It wokrs almost never when passed through an intent. What is the relevant difference here?

Unrelated but interesting anyway: https://www.javacodegeeks.com/2014/06/javamail-can-be-evil-and-force-you-to-restart-your-app-server.html
